tests/unit/test_index.py

```diff
@@ -159,12 +159,14 @@ def test_get_formatted_locations_basic_auth():
         'https://pypi.org/simple',
         'https://user:pass@repo.domain.com',
     ]
-    finder = PackageFinder([], index_urls, session=[])
+    find_links = [
+        'https://user:pass@page.domain.com'
+    ]
+    finder = PackageFinder(find_links, index_urls, session=[])
 
     result = finder.get_formatted_locations()
-    assert 'user' in result
-    assert '****' in result
-    assert 'pass' not in result
```

Comment:
This should be kept (and duplicated for `find_links`, maybe with a `pass_index` and a `pass_find_links`.